lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -106,1360 +56,800 @@ function Install-Prerequisites {
-    $fileExists = Test-Path $download_location
-    if ($fileExists){
-        if ($md5){
-            $fileHash = (Get-FileHash -Algorithm MD5 $download_location).Hash
-            if ($fileHash -eq $md5){
-                return $download_location
-            }
-        }else{
-            return $download_location
+function Start-InstallHook {
+    if (!(Get-IsNanoServer)) {
+        try {
+            Set-MpPreference -DisableRealtimeMonitoring $true
+        } catch {
+            # No need to error out the hook if this fails.
+            Write-JujuWarning "Failed to disable antivirus: $_"
         }
     }
-    Write-JujuInfo "Downloading file from $url to $download_location"
+    # Set machine to use high performance settings.
     try {
-        Start-ExecuteWithRetry {
-            Invoke-FastWebRequest -Uri $url -OutFile $download_location
-        }
+        Set-PowerProfile -PowerProfile Performance
     } catch {
-        Write-JujuErr "Could not download $url to destination $download_location"
-        Throw
-    }
-    return $download_location
-}
-
-function Get-NovaInstaller {
-    $distro = Get-OpenstackVersion
-    $installer_url = Get-JujuCharmConfig -Scope "installer-url"
-    if ($distro -eq $false){
-        $distro = "liberty"
-    }
-    Write-JujuInfo "installer-url is set to: $installer_url"
-    if (!$installer_url) {
-        if (!$distro_urls[$distro] -or !$distro_urls[$distro]["installer"]){
-            Throw "Could not find a download URL for $distro"
-        }
-        if ((Get-IsNanoServer))  {
-            if (!$distro_urls[$distro]["installer"]["zip"]) {
-                Throw "Distro $distro does not support Nano server"
-            }
-            $url = $distro_urls[$distro]["installer"]["zip"]
-        } else {
-            $url = $distro_urls[$distro]["installer"]["msi"]
-        }
-    }else {
-        $url = $installer_url
-    }
-    [string]$location = Download-File $url
-    return $location
-}
-
-function Install-NovaFromMSI {
-    param(
-        [Parameter(Mandatory=$true)]
-        [string]$InstallerPath
-    )
-    Write-JujuInfo "Running Nova install"
-    $hasInstaller = Test-Path $InstallerPath
-    if($hasInstaller -eq $false){
-        $InstallerPath = Get-NovaInstaller
-    }
-    Write-JujuInfo "Installing from $InstallerPath"
-    $ret = Start-Process -FilePath msiexec.exe -ArgumentList "SKIPNOVACONF=1","INSTALLDIR=`"$installDir`"","/qn","/l*v","$env:APPDATA\log.txt","/i","$InstallerPath" -Wait -PassThru
-
-    if($ret.ExitCode) {
-        Throw ("Failed to install Nova: {0}" -f $ret.ExitCode)
+        # No need to error out the hook if this fails.
+        Write-JujuWarning "Failed to set power scheme."
     }
-    return $true
-}
+    Start-TimeResync
 
-function Install-NovaFromZip {
-    param(
-        [Parameter(Mandatory=$true)]
-        [string]$InstallerPath
-    )
-    $template_dir = Get-TemplatesDir
-    $distro = Get-OpenstackVersion
+    $renameReboot = Rename-JujuUnit
```

Comment:
We never do this on MAAS machines, unless you explicitly set `change-hostname` to true (default is false).